### PR TITLE
Removed HTTP support from DPS device client

### PR DESCRIPTION
**Pull request by tmahmood-microsoft**
- Removed support for HTTP in DPS device client.
- [minor] Using latest API version for EdgeModuleClientHelper

### CODE REVIEW

File: SDK v2 migration guide.md [hunk 1]
```diff
@@ -175,6 +175,7 @@ to migrate to version 2.x when they have the chance. For more details on LTS rel
 - The security providers that are used in conjunction with this client have changed. See [this section](#security-provider-clients) for more details.
 - The previous way of providing transport level settings (`ProvisioningTransportHandler`) has been replaced with `ProvisioningClientTransportSettings`.
 - TPM support removed. The library used for TPM operations is broken on Linux and support for it is being shutdown. We'll reconsider how to support HSM.
+- HTTP has been removed as a transport option to keep the provisioning SDK consistent with IoT hub device SDK.
```
Comment (timtay-microsoft) [inline SDK v2 migration guide.md:178]:
```suggestion
- HTTP has been removed as a transport option to keep the provisioning device SDK consistent with IoT hub device SDK.
```

File: provisioning/device/src/Transports/Amqp/ClientWebSocketTransport.cs [hunk 2]
```diff
@@ -88,10 +88,6 @@ private async Task<bool> WriteAsyncCoreAsync(TransportAsyncCallbackArgs args)
             {
                 throw new IOException(webSocketException.Message, webSocketException);
             }
-            catch (HttpListenerException httpListenerException)
```
Comment (timtay-microsoft) [inline provisioning/device/src/Transports/Amqp/ClientWebSocketTransport.cs:91]: This change is in the AMQP stack, FYI. I don't think we need to remove this catch

**Review by timtay-microsoft — APPROVED** [review]

File: provisioning/device/src/Utilities/ClientApiVersionHelper.cs [hunk 3]
```diff
@@ -5,7 +5,7 @@ namespace Microsoft.Azure.Devices.Provisioning.Client
 {
     internal class ClientApiVersionHelper
     {
-        // TODO: Split ApiVersionName as it is only used by HTTP and AMQP.
+        // TODO: Split ApiVersionName as it is only used by AMQP.
```
Comment (drwill-ms) [inline provisioning/device/src/Utilities/ClientApiVersionHelper.cs:8]: Why not MQTT?

**Review by drwill-ms — APPROVED** [review]

**Pull request merged**